librz/core/rz_basefind.cpp

```diff
@@ -0,0 +1,177 @@
+//MIT Copyright (c) 2013 Michael Coppola
+//MIT Copyright (c) 2021 Oleg Bushin
+
+//TODO Rewrite code from C++ to C.
+//Can be used for example rizan structures :RzVector, RzPVector, RzList, RzBase (have methods to return an RzList<RzBase>).
+// You can also use hashmaps (since have std::pair in the c++ code) for those one,
+// you need HtPP or HtUP or HtUU or HtPU where P is void* and U is uint64_t.
+// To open/read/etc the file you can use <rz_io.h>.
+// in these files:
+//      rizin/meson.build {
+//          project('rizin', 'cpp', 'c', ...) - delete 'cpp'
+//          add_project_arguments('-lstdc++', language: 'c')  - delete line
+//      rizin/librz/meson.build
+//          add_project_arguments('-lstdc++', language: 'c')  - delete line
+//      rizin/librz/core/meson.build {
+//          rz_core_sources = ['rz_basefind.cpp', ...] - replace by 'rz_basefind.c'
+//TODO Implement multithreading.
+
+#include "rz_basefind.h"
+#include <algorithm>
+#include <climits>
+#include <cstdint>
+#include <cstring>
+#include <fstream>
+#include <map>
+#include <unordered_set>
+#include <vector>
+#include <iostream>
+static const size_t minLength = 10;
+static const size_t WORD_LEN = 4; //32-bit assumed in a few places below
+typedef uint32_t offset_t;
+typedef std::vector<std::pair<offset_t, unsigned>> ptable; //pointer table
+typedef std::unordered_set<offset_t> stable; //string table
+typedef std::vector<std::pair<offset_t, unsigned>> scorecard;
```

Comment:
for this you can use
```c
typedef rz_basefinds_t {
     offset_t offset;
     ut32 score;
} RzBasefinds;
```
The all `std::vector` and `std::unordered_set` you can use `rz_pvector` and use the `rz_pvector_find` to check if the set has the address.